doc/aaa/radius_authentication.md

```diff
@@ -0,0 +1,634 @@
+                                 ; outgoing UDP/IP dgram. Default is
+                                 ; determined by routing stack
+                                 ; (WIP) RADIUS_SERVER attribute is preferred.
+```
+
+
+### RADIUS_SERVER Table Schema
+
+```
+; RADIUS per server configuration in the system.
+; Key
+server_key           = NameOrIPAddress; RADIUS server's DNS name or IPv4|6 addr
+                                      ; * DNS name is Work In Progress(WIP) *
+                                      ; * in future phase                   *
+; Attributes
+auth_port            = 1*5DIGIT
+passkey              = 1*32VCHAR  ; per server shared secret (Valid chars: ASCII printable except SPACE, '#', and COMMA)
+auth_type            = "pap" / "chap" / "mschapv2"  ; method used for authenticating the communication message
+priority             = 1*2DIGIT  ; specify RADIUS server's priority
+timeout              = 1*2DIGIT
+retransmit           = 1*2DIGIT
+vrf                  = vrf_name  ; Use default vrf if not specified
+src_intf             = source_interface ; Eg: eth0, Loopback0... Use the first
+                                 ; IPAddress retrieved for interface for the
+                                 ; outgoing UDP/IP dgram. Default is
+                                 ; determined by routing stack
+```
+
+## Manageability
+
+### Data Models
+
+For configuring RADIUS, we will use [Openconfig AAA RADIUS](https://github.com/openconfig/public/blob/master/release/models/system/openconfig-aaa-radius.yang)
+Only certain config related parts will be implemented. The supported config
+paths are:
+
+- /oc-sys:system/oc-aaa:aaa/oc-aaa:server-groups/oc-aaa:server-group[name=RADIUS]/oc-aaa:servers/oc-aaa:server[address=IPAddress]/config
+
+  The container's oc-aaa:acct-port leaf will not be  supported.
+
+  The container will be augmented to support the following additional leaves:
+
+  - auth-type
+  - priority
+  - vrf
+  - source-interface
+
+  The oc-aaa:source-address will be supported only for global RADIUS conf.
+
+  The oc-aaa:auth-port leaf will be supported only for per server conf.
+
+- /oc-sys:system/oc-aaa:aaa/oc-aaa:server-groups/oc-aaa:server-group[name=RADIUS]/config
+
+  The container will be augmented to support the following additional leaves
+  for global RADIUS configuration:
+
+  - source-address
+  - auth-type
+  - secret-key
+  - timeout
+  - retransmit-attempts
+  - nas-ip-address         (WIP in future phase)
+  - statistics             (WIP in future phase)
+
+### CLI
+
+The RADIUS commands will be implemented in click, and KLISH using the available
+CLI framework.
+
+For TACACS+, the existing module click comands will continue to be available.
+The AAA commands are an existing click module command which will be extended
+for the RADIUS Management User Authentication feature.
+
+
+```
+sonic# show aaa
+sonic# show radius
+
+sonic(config)# [no] radius-server source-ip <IPAddress(IPv4 or IPv6)>
```

Comment:
I am unable to understand "sonic(config)# [no]"

Also can you paste a sample o/p for the show command

Reply:
The optional "no" prefix to most commands is a way to un-configure the corresponding configuration. I will add something like this to the HLD to give a better picture of the CLI.

```
S5(config)# radius-server nas-ip 10.59.143.170
S5(config)# radius-server host 10.59.143.220
S5(config)# do show radius-server
---------------------------------------------------------
RADIUS Global Configuration
---------------------------------------------------------
nas-ip-addr: 10.59.143.170
--------------------------------------------------------------------------------
HOST            AUTH-TYPE KEY       AUTH-PORT PRIORITY TIMEOUT RTSMT VRF   SI 
--------------------------------------------------------------------------------
10.59.143.220   -         -         1812      1        -       -     -     - 
S5(config)# no radius-server nas-ip 
S5(config)# do show radius-server
---------------------------------------------------------
RADIUS Global Configuration
---------------------------------------------------------
--------------------------------------------------------------------------------
HOST            AUTH-TYPE KEY       AUTH-PORT PRIORITY TIMEOUT RTSMT VRF   SI 
--------------------------------------------------------------------------------
10.59.143.220   -         -         1812      1        -       -     -     - 
S5(config)# 
```